05-session-kit-transact/index.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    Action,
+    Asset,
+    Name,
+    Session,
+    Struct,
+    WalletPluginPrivateKey,
+} from '@[SOMETHING]kit/session'
+
+// Instantiate a session to a blockchain account
+const session = new Session({
+    // Pass information about the blockchain and how to connect to it.
+    chain: {
+        id: '2a02a0053e5a8cf73a56ba0fda11e4d92e0238a4a2aa74fccf46d5a910746840',
+        url: 'https://jungle3.greymass.com',
+    },
+    // Optionally pass APIClient to override the one automatically created.
+    // client: new APIClient({url: 'https://jungle3.greymass.com'}),
+    // Pass the permission to use for this session.
+    permissionLevel: 'someone@active',
+    // Pass a Wallet Plugin that this session will interact with.
+    walletPlugin: new WalletPluginPrivateKey({
+        privateKey: '5JnUd2V5nYmRKgK9K2fRQcs3qKoi4mbcGV8Dg8EFqjjqEp9tYP5',
+    }),
+})
+
+// -----------------------------------------------------------------
+// --- BOILERPLATE TO SETUP TRANSACTION, NOT PART OF SESSION KIT ---
+// -----------------------------------------------------------------
+
+// Setup core struct for the action to be performed, to avoid get_abi call
+@Struct.type('transfer')
+class Transfer extends Struct {
```

Comment:
Ideally this structure and a method to call it would be provided by the Contract Kit, right?

Reply:
I just simplified the example to show without this.